nautobot/users/filters.py

```diff
@@ -69,6 +69,34 @@ def search(self, queryset, name, value):
         )
 
 
+class TokenFilterSet(BaseFilterSet):
+    q = django_filters.CharFilter(
+        method="search",
+        label="Search",
+    )
+    user_id = django_filters.ModelMultipleChoiceFilter(
+        field_name="user",
+        queryset=get_user_model().objects.all(),
+        label="User",
+    )
+    user = django_filters.ModelMultipleChoiceFilter(
+        field_name="user__username",
+        queryset=get_user_model().objects.all(),
+        to_field_name="username",
+        label="User (name)",
+    )
+    created = django_filters.DateTimeFilter()
+    created__gte = django_filters.DateTimeFilter(field_name="created", lookup_expr="gte")
+    created__lte = django_filters.DateTimeFilter(field_name="created", lookup_expr="lte")
+    expires = django_filters.DateTimeFilter()
+    expires__gte = django_filters.DateTimeFilter(field_name="expires", lookup_expr="gte")
+    expires__lte = django_filters.DateTimeFilter(field_name="expires", lookup_expr="lte")
```

Comment:
These special lookups should not be needed so long as the created and updated fields are added to the meta field list. These additional lookup fields will be autogenerated by the BaseFilterSet.

Reply:
@lampwins Removing manual definitions and adding them to the meta fields and letting the autogeneration happen makes the unit tests break...

Code: https://github.com/imranh2/nautobot/commit/803680454bcdef47e3d45f5993bb0b51fb2d08f8#diff-65bbd5f28c2b762b53e2f7552cb90da33f364fc77ccfcd17ae1a90ad555ed1a6R91
Tests: https://github.com/imranh2/nautobot/runs/5394168838?check_suite_focus=true

Returns 3 tokens instead of 2, so not evaluating the date (datetime.datetime(2000, 1, 1)) correctly...

Reply:
If I define `expires` then `expires__lte` fails. I'm clearly using this wrong but don't know how I should be using it...

Reply:
Searched the code for an example and have failed to find it. The following is what I found and it doesn't use the auto-generation stuff.

```
$ grep -rni "create*.__" .
./nautobot/extras/tests/test_api.py:369:        response = self.client.get("{}?created__gte=2001-02-04".format(url), **self.header)
./nautobot/extras/tests/test_api.py:377:        response = self.client.get("{}?created__lte=2001-02-04".format(url), **self.header)
./nautobot/extras/filters.py:88:    created__gte = django_filters.DateFilter(field_name="created", lookup_expr="gte")
./nautobot/extras/filters.py:89:    created__lte = django_filters.DateFilter(field_name="created", lookup_expr="lte")
```

@lampwins any advice?